### PR TITLE
[SCRP-32] com client

### PR DESCRIPTION
**Pull request by pablo-lo**
This PR implements parts of ticket 32 of the CPR project.

### CODE REVIEW

File: api/users.go [hunk 1]
```diff
@@ -9,17 +9,20 @@ import (
...
-	executeActionsEmailPath    = userIDPath + "/execute-actions-email"
-	sendReminderEmailPath      = "/auth/realms/:realm/onboarding/sendReminderEmail"
-	smsAPI                     = "/auth/realms/:realm/smsApi"
-	sendSmsCode                = smsAPI + "/sendNewCode"
-	shadowUser                 = userIDPath + "/federated-identity/:provider"
+	userPath                       = "/auth/admin/realms/:realm/users"
+	adminExtensionAPIPath          = "/auth/realms/:realmReq/api/admin/realms/:realm"
+	usersAdminExtensionAPIPath     = adminExtensionAPIPath + "/users"
+	sendEmailAdminExtensionAPIPath = adminExtensionAPIPath + "/send-email"
+	sendSMSAdminExtensionAPIPath   = adminExtensionAPIPath + "/send-sms"
```
Comment (fperot74) [inline api/users.go:16]: Specifications in JIRA ticket were not correctly defined:
POST /auth/realms/<realm>/sendSms

File: api/users.go [hunk 2]
```diff
@@ -124,3 +127,15 @@ func (c *Client) LinkShadowUser(accessToken string, reqRealmName string, userID
 	_, err := c.post(accessToken, nil, url.Path(shadowUser), url.Param("realm", reqRealmName), url.Param("id", userID), url.Param("provider", provider), body.JSON(fedIDKC))
 	return err
 }
+
+// SendCPREmail sends an email to a user in the format definden in the CPR project
```
Comment (fperot74) [inline api/users.go:131]: This method is not supposed to be project specific. Please remove references to this project.

File: api/users.go [hunk 3]
```diff
@@ -124,3 +127,15 @@ func (c *Client) LinkShadowUser(accessToken string, reqRealmName string, userID
...
+func (c *Client) SendCPREmail(accessToken string, reqRealmName string, realmName string, cprEmailRep keycloak.CPREmailRepresentation) error {
```
Comment (harture) [inline api/users.go:132]: There is nothing specific to CPR. So we can have a generic comment and method name

File: api/users.go [hunk 4]
```diff
@@ -124,3 +127,15 @@ func (c *Client) LinkShadowUser(accessToken string, reqRealmName string, userID
...
+	_, err := c.post(accessToken, nil, url.Path(sendEmailAdminExtensionAPIPath), url.Param("realmReq", reqRealmName), url.Param("realm", realmName), body.JSON(cprEmailRep))
+	return err
+}
+
+// SendCPRSMS sends an SMS to a user in the format definden in the CPR project
+func (c *Client) SendCPRSMS(accessToken string, reqRealmName string, realmName string, cprSMSRep keycloak.CPRSMSRepresentation) error {
```
Comment (harture) [inline api/users.go:138]: Same here nothing specific to CPR

File: api/users.go [hunk 5]
```diff
@@ -124,3 +127,15 @@ func (c *Client) LinkShadowUser(accessToken string, reqRealmName string, userID
...
 	return err
 }
+
+// SendCPREmail sends an email to a user in the format definden in the CPR project
+func (c *Client) SendCPREmail(accessToken string, reqRealmName string, realmName string, cprEmailRep keycloak.CPREmailRepresentation) error {
+	_, err := c.post(accessToken, nil, url.Path(sendEmailAdminExtensionAPIPath), url.Param("realmReq", reqRealmName), url.Param("realm", realmName), body.JSON(cprEmailRep))
+	return err
+}
+
+// SendCPRSMS sends an SMS to a user in the format definden in the CPR project
```
Comment (fperot74) [inline api/users.go:137]: This method is not supposed to be project specific. Please remove references to this project.

File: definitions.go [hunk 6]
```diff
@@ -757,3 +757,45 @@ type RecoveryCodeRepresentation struct {
 type ActivationCodeRepresentation struct {
 	Code *string `json:"code,omitempty"`
 }
+
+// CPREmailRepresentation struct
```
Comment (harture) [inline definitions.go:761]: Remove all CPR specific stuff.

Comment (fperot74) [inline definitions.go:761]: Remove all references to CPR

File: definitions.go [hunk 7]
```diff
@@ -757,3 +757,45 @@ type RecoveryCodeRepresentation struct {
...
+	Theming     *CPREmailThemingRepresentation  `json:"theming,omitempty"`
+	Attachments *[]CPRAttachementRepresentation `json:"attachments,omitempty"`
+}
+
+// CPREmailThemingRepresentation struct
+type CPREmailThemingRepresentation struct {
+	SubjectKey         *string                              `json:"subjectKey,omitempty"`
+	SubjectParameters  *[]string                            `json:"subjectParameters,omitempty"`
+	Template           *string                              `json:"template,omitempty"`
+	TemplateParameters *CPRTemplateParametersRepresentation `json:"templateParameters,omitempty"`
```
Comment (fperot74) [inline definitions.go:773]: Remove this TemplateParametersRepresentation type and simply use a map[string]string

**Review by fperot74 — CHANGES_REQUESTED** [review]

**Review by harture — APPROVED** [review]

**Review by fperot74 — APPROVED** [review]
Please squash and reword your commit, something like this: [SCPR-32] Add methods to send email and SMS
We may wait to have tested from end to end (keycloak-bridge/keycloak-client/keycloak) before merging

**Pull request merged**